src/ssl_load.c

```diff
@@ -2803,9 +2803,27 @@ int wolfSSL_CTX_load_verify_locations_ex(WOLFSSL_CTX* ctx, const char* file,
         }
 
         if (file != NULL) {
+        #if defined(OPENSSL_EXTRA) && defined(HAVE_CRL)
+            /* handle CRL type being passed in */
+            WOLFSSL_CRL crl;
+
+            XMEMSET(&crl, 0, sizeof(WOLFSSL_CRL));
+            crl.cm = ctx->cm;
+
             /* Load the PEM formatted CA file. */
+            ret = ProcessFile(ctx, file, WOLFSSL_FILETYPE_PEM, DETECT_CERT_TYPE,
+                NULL, 0, &crl, verify);
+            /* found a good CRL, add it to ctx->cm */
+            if (ret == 1 && crl.crlList != NULL) {
+                wolfSSL_X509_STORE_add_crl(wolfSSL_CTX_get_cert_store(ctx),
+                    &crl);
+            }
+        #else
+            /* Load the PEM formatted CA file. No CRL support, only expecting
+             * CA's */
             ret = ProcessFile(ctx, file, WOLFSSL_FILETYPE_PEM, CA_TYPE, NULL, 0,
                 NULL, verify);
+        #endif
```

Comment:
I don't see anything in https://docs.openssl.org/master/man3/SSL_CTX_load_verify_locations about needing to support CRL's here.

Reply:
This one surprised me. It looks like it was handling a PEM file that has a CRL header though. Test case it was done for is here (https://github.com/python/cpython/blob/960936fe90f9a0b947c7ed10f91582ff6664c0b8/Lib/test/test_ssl.py#L2932)

Reply:
I'm not convinced this is correct. Please go through OpenSSL to see if this is intentional or just coincidence that it works.

Reply:
In OpenSSL, function trace:
`SSL_CTX_load_verify_locations` -> `SSL_CTX_load_verify_file` -> `X509_STORE_load_file_ex` -> `X509_LOOKUP_load_file_ex(l, f, X509_FILETYPE_PEM, lib, arg)`

At that point it's treating it as a simple X509_FILETYPE_PEM file being passed in. I think it's something that is a coincidence that it works. Regardless of that though there is a Python test expecting the behavior. I can limit this to the WOLFSSL_PYTHON macro.

Reply:
It was added here and its definitely an error. https://github.com/python/cpython/commit/225877917e002df4b2d87e965ddd30226aa209ec#diff-d526ded1c360bed6b222de46f4ca92b834f978ebed992fb3189bf9a94a198578R1826

Reply:
Fixed by adding a case to the Python patch for it to fall back and try loading it as a CRL if the call to load verify locations fails.